ext/mbstring/libmbfl/filters/mbfilter_utf8.c

```diff
@@ -335,3 +337,17 @@ static void mb_wchar_to_utf8(uint32_t *in, size_t len, mb_convert_buf *buf, bool
 
 	MB_CONVERT_BUF_STORE(buf, out, limit);
 }
+
+static zend_string* mb_cut_utf8(unsigned char *str, size_t from, size_t len, unsigned char *end)
+{
+	unsigned char *start = str + from;
+	while (start > str && ((signed char)*start) < -64)
```

Comment:
I don't understand why there is a check for ``< -64`` ?

Reply:
That's a fast way to tell if a UTF-8 byte is a continuation byte (i.e. 2nd, 3rd, or 4th byte of a multi-byte character).

I'll comment it.

Reply:
Ah right this makes sense now that you explain it :D

Reply:
Isn't it the same as 
```
((unsigned char)*start) < 192
```

Reply:
@kamil-tekiela You actually need to use signed comparisons, not unsigned comparisons, for this trick to work. It doesn't work at all with an unsigned comparison.

The reason is because the "wraparound point" from lowest number to highest number is different for signed and unsigned numbers. For unsigned, 00000000 is lowest and 11111111 is highest. For unsigned, 10000000 is lowest and 01111111 is highest.